File: 백준/Silver/1874. 스택 수열/스택 수열.cc
#include <iostream>
#include <vector>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    
    int n;
    cin >> n;
    
    int current = 1;
    bool is_possible = true;
    vector<int> stack;
    vector<char> ans;
    for (int i = 0; i < n; i++) {
        int num;
        cin >> num;
        
        while(current <= num) {
            stack.push_back(current++);
            ans.push_back('+');
        }
        if (stack.empty() || num != stack.back()) {
            is_possible = false;
            break;
        }
        stack.pop_back();
        ans.push_back('-');
    }
    if (is_possible) {
        for (char c: ans) {
            cout << c << "\n";
        }
    } else {
        cout << "NO";
    }
    return 0;
}

File: 백준/Silver/1874. 스택 수열/스택 수열.py
import sys

input = sys.stdin.readline

n = int(input())
current = 1

stack = []
ans = []
is_possible = True

for _ in range(n):
    num = int(input())
    if current > num:
        if num == stack[-1]:
            stack.pop()
            ans.append("-")
        else:
            is_possible = False
            break
    else:
        while(current <= num):
            stack.append(current)
            ans.append("+")
            current += 1
        
        stack.pop()
        ans.append("-")
            
if is_possible:
    print("\n".join(ans))
else:
    print("NO")
        
        